src/TMap.cpp

```diff
@@ -3080,10 +3080,11 @@ std::pair<bool, QString> TMap::readJsonMapFile(const QString& source, const bool
         TArea* pArea = new TArea(this, pNewRoomDB);
```

Comment:
In modern C++ it would be a lot better to use `std::unique_ptr` - then this bug would never happen, since the area would automatically get deleted once it goes out of scope.

Reply:
We don't want it to - if the load is NOT aborted it gets added into the new `TRoomDB` instance and we do not want the pointed to `TArea` destroyed then...

Reply:
That's when you use `release()` which clearly communicates what you're doing ;) https://en.cppreference.com/w/cpp/memory/unique_ptr/release

Reply:
I've made a version that works here, check it out: https://github.com/Mudlet/Mudlet/pull/5129